internal/pkg/manifest/svc_test.go

```diff
@@ -147,7 +147,7 @@ environments:
 						Sidecars: map[string]*SidecarConfig{
 							"xray": {
 								Port:       aws.String("2000/udp"),
-								Image:      aws.String("123456789012.dkr.ecr.us-east-2.amazonaws.com/xray-daemon"),
+								Image:      BasicToUnion[*string, SidecarImageConfig](aws.String("123456789012.dkr.ecr.us-east-2.amazonaws.com/xray-daemon")),
```

Comment:
It'd be amazing if we can add a test case here where the `inContent` has the advanced configuration for the sidecar and we ensure that we're unmarshaling properly into `SidecarConfig`